light-client/relayer/src/relayer.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) 2018-2023 The MobileCoin Foundation
+
+use crate::{error::Error, verifier::Verifier, Config, Sender};
+use mc_account_keys::burn_address_view_private;
+use mc_blockchain_types::{Block, BlockContents, BlockData, BlockIndex, BlockMetadata};
+use mc_common::logger::{log, Logger};
+use mc_ledger_db::{Error as LedgerError, Ledger, LedgerDB};
+use mc_transaction_core::tx::TxOut;
+use mc_util_telemetry::{block_span_builder, telemetry_static_key, tracer, Key, Span, Tracer};
+use mc_watcher::{error::WatcherDBError, watcher_db::WatcherDB};
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex, MutexGuard,
+    },
+    thread::{Builder as ThreadBuilder, JoinHandle},
+    time::Duration,
+};
+
+/// Telemetry: block index currently being worked on.
+const TELEMETRY_BLOCK_INDEX_KEY: Key = telemetry_static_key!("block-index");
+
+#[derive(Default, Debug, Clone)]
+pub struct BurnTx {
```

Comment:
This name is suboptimal since in theory there could be multiple burn TxOuts inside the block, and its also possible that we will be relaying other blocks (e.g. blocks containing mint transactions) so I would rename this to `RelayedBlock` and change the `tx_outs` field to `burn_tx_outs`